tests/unit/test_charm.py

```diff
@@ -0,0 +1,170 @@
+    @patch(
+        "charm.PrometheusConfigurerOperatorCharm._prometheus_server_host",
+        new_callable=PropertyMock,
+    )
+    @patch(
+        "charm.PrometheusConfigurerOperatorCharm._prometheus_server_port",
+        new_callable=PropertyMock,
+    )
+    @patch("charm.AlertRulesDirWatcher", Mock())
+    def test_given_prometheus_relation_created_and_prometheus_configurer_container_ready_when_pebble_ready_then_pebble_plan_is_updated_with_correct_pebble_layer(  # noqa: E501
+        self,
+        patched_prometheus_server_port,
+        patched_prometheus_server_host,
+        patched_prometheus_configurer_port,
+        patched_rules_dir,
+    ):
+        test_prometheus_server_port = 4321
+        test_prometheus_server_host = "testhost"
+        test_prometheus_configurer_port = 1234
+        test_rules_dir = "/test/rules/dir"
+        patched_prometheus_server_port.return_value = test_prometheus_server_port
+        patched_prometheus_server_host.return_value = test_prometheus_server_host
+        patched_prometheus_configurer_port.return_value = test_prometheus_configurer_port
+        patched_rules_dir.return_value = test_rules_dir
+        self.harness.add_relation("prometheus", "prometheus-k8s")
+        expected_plan = {
+            "services": {
+                "prometheus-configurer": {
+                    "override": "replace",
+                    "startup": "enabled",
+                    "command": "prometheus_configurer "
+                    f"-port={test_prometheus_configurer_port} "
+                    f"-rules-dir={test_rules_dir}/ "
+                    f"-prometheusURL={test_prometheus_server_host}:{test_prometheus_server_port} "
+                    f"-multitenant-label={TEST_MULTITENANT_LABEL} "
+                    "-restrict-queries",
+                }
+            }
+        }
+        self.harness.container_pebble_ready("prometheus-configurer")
+
+        updated_plan = self.harness.get_container_pebble_plan("prometheus-configurer").to_dict()
+        self.assertEqual(expected_plan, updated_plan)
+
+    def test_given_dummy_http_server_container_ready_when_pebble_ready_then_pebble_plan_is_updated_with_correct_pebble_layer(  # noqa: E501
+        self,
+    ):
+        expected_plan = {
+            "services": {
+                "dummy-http-server": {
+                    "override": "replace",
+                    "startup": "enabled",
+                    "command": "nginx",
+                }
+            }
+        }
+        self.harness.container_pebble_ready("dummy-http-server")
+
+        updated_plan = self.harness.get_container_pebble_plan("dummy-http-server").to_dict()
+        self.assertEqual(expected_plan, updated_plan)
+
+    @patch("charm.PrometheusConfigurerOperatorCharm.model")
+    @patch("charm.AlertRulesDirWatcher", Mock())
+    def test_given_prometheus_relation_created_and_prometheus_configurer_container_ready_when_pebble_ready_then_charm_goes_to_active_state(  # noqa: E501
```

Comment:
instead of patching that the relations were created, you can create them with the `add_relation` from the harness.

Reply:
I know, I did that in one of the earlier tests. I figured that since the relation being created is the precondition for this test I may as well patch it as long as at least one test in the suite actually creates the relation to validate logic handling relation.
After you mentioned it, I think above approach may be confusing, so I removed the patch.